pyproject_metadata/__init__.py

```diff
@@ -195,9 +195,12 @@ class StandardMetadata:
     dynamic: list[str] = dataclasses.field(default_factory=list)
 
     def __post_init__(self) -> None:
-        self.name = re.sub(r'[-_.]+', '-', self.name).lower()
         self._update_dynamic(self.version)
 
+    @property
+    def canonnical_name(self) -> str:
```

Comment:
```suggestion
    def canonical_name(self) -> str:
```

Tweaking the spelling here.